src/states/adult.rs

```diff
@@ -357,6 +357,14 @@ impl Approved for Adult {
         &mut self.chain
     }
 
+    fn set_pfx_successfully_polled(&mut self, _: bool) {
+        // Doesn't do anything
+    }
+
+    fn is_pfx_successfully_polled(&self) -> bool {
+        false
```

Comment:
will have to think about this. right now an adult is someone staying in this state until they're established, so they certainly do go past the Genesis gossip stage, so this returning blanket false seems weird. Although this cache logic and how much we need adults to also handle this I need to think about too.